aiida_lammps/workflows/base.py

```diff
@@ -0,0 +1,351 @@
+            self.exposed_inputs(LammpsBaseCalculation, "lammps")
+        )
+        self.ctx.inputs.settings = (
+            self.ctx.inputs.settings.get_dict() if "settings" in self.ctx.inputs else {}
+        )
+
+    def validate_parameters(self):
+        """Validate the parameters for the workchain"""
+
+        if "store_restart" in self.ctx.inputs:
+            self.ctx.inputs.settings.store_restart = self.ctx.inputs.store_restart.value
+
+        if "script" not in self.ctx.inputs and any(
+            key not in self.ctx.inputs
+            for key in ["parameters", "potential", "structure"]
+        ):
+            raise InputValidationError(
+                "Unless `script` is specified the inputs `structure`, "
+                "`potential` and  `parameters` have to be specified."
+            )
+
+        if "parameters" in self.ctx.inputs:
+            self.ctx.inputs.parameters = self.ctx.inputs.parameters.get_dict()
+
+            if not any(key in self.ctx.inputs.parameters for key in ["md", "minimize"]):
+                raise InputValidationError(
+                    "If not using a script the type of calculation, either "
+                    "'md' or 'minimize' must be given"
+                )
+
+            _restart = any(
+                self.ctx.inputs.parameters.get("restart", {}).get(key, False)
+                for key in ["print_final", "print_intermediate"]
+            )
+            _store_restart = (
+                "store_restart" in self.ctx.inputs
+                and self.ctx.inputs.store_restart.value
+            )
+
+            if _store_restart and not _restart:
+                raise InputValidationError(
+                    "To store the restartfile one needs to indicate "
+                    "that either the final or intermediate restartfiles must be printed"
+                )
+
+    def report_error_handled(self, calculation, action):
+        """Report an action taken for a calculation that has failed.
+
+        This should be called in a registered error handler if its condition
+        is met and an action was taken.
+        :param calculation: the failed calculation node
+        :param action: a string message with the action taken
+        """
+        self.report(
+            f"{calculation.process_label}<{calculation.pk}> failed "
+            f"with exit status {calculation.exit_status}: {calculation.exit_message}"
+        )
+        self.report(f"Action taken: {action}")
+
+    def _check_restart_in_remote(self, calculation):
+        """
+        Check if the remote folder of a previous calculation contains a restartfile
+
+        :param calculation: node from the previous calculation
+        :return: latest restartfile found or None
+        """
+
+        _restartfile = calculation.get_metadata_inputs()["restart_filename"]
+
+        restartfiles = [
+            entry
+            for entry in calculation.outputs.remote_folder.listdir()
```

Comment:
There are two potential problems with using `RemoteData.listdir` as it will open a transport to the remote computer:

1. The opening of the transport can fail (for example connection problems) and the exceptions are not caught here, which therefore will except the entire workchain, making this approach very vulnerable to connection problems.
2. The opening of the transport is done directly and not through the transport pool that is maintained by the engine for `CalcJobs`. This means that each workchain will open its own connection. If you run many workchains, that means you can hit the remote server too often and get banned.

Unfortunately, there is no API that allows to reuse all the engine's safety mechanisms surrounding transports that is used for `CalcJobs`, here in `WorkChains`. If possible, I would strongly recommend not using this approach, but if there is no other option, then we should maybe document this so that people are aware of these pitfalls.

Reply:
Yes, this is a tricky one. 
If you look in the calculation I do a similar check
```python


        # check if there is a parent folder to restart the simulation from a previous run
        if "parent_folder" in self.inputs:
            # Check if one should do symlinks or if one should copy the files
            # By default symlinks are used as the file can be quite large
            symlink = settings.pop("parent_folder_symlink", self._default_symlink_usage)
            # Find the name of the previous restartfile, if none is given the default one is assumed
            # Setting the name here will mean that if the input file is generated from the parameters
            # that this name will be used
            _read_restart_filename = settings.pop(
                "previous_restartfile", self._DEFAULT_VARIABLES.get("restart_filename")
            )

            if not _read_restart_filename in self.inputs.parent_folder.listdir():
                raise exceptions.InputValidationError(
                    f'The name "{_read_restart_filename}" for the restartfile is not present in the '
                    f'remote folder "{self.input.parent_folder.uuid}"'
                )
                ```

I have no idea of how else to check if the file is present in the remote. Because that is basically what this is trying to do, check if the file exists in the parent folder. In the case of the `RestartWorkChain` of course it is a bit trickier since we also need to find the name of the `restartfile`.  Why one needs to do this is due to the fact that you can write the restartfile at regular intervals during your calculation, this will then write files that look like `{BASE_RESTARTFILE_NAME}.{step}`. Hence, you have the problem that you do not know exactly how your restartfile will be named, and then you need to check.

The only way I can possibly think of solving this, is to somehow store this information somewhere in the parsing of the calculation, since if you give the instruction that those intermediate restartfiles are printed they are placed in the `retrieve_temporary_list`. Then you could find out what is the name of the restartfile, if present, and then add it somewhere in the parsing output. Maybe in the `results` dictionary? I think that one should add a section there with `WARNINGS` and so on, so maybe there? 

What do you think @sphuber ? That would mean that we know the name, we would however not check for its existence. But maybe that is an acceptable compromise